### ISSUE TITLE
Unable to generate correct Prisma Error Code for Required Fields

### ISSUE THREAD

**Issue by arkmech** [prisma/prisma#15829]
### Bug description

Goal: Want to receive the proper Prisma error code with meta that includes target field data to throw my appropriate errors based on missing required field.

Per Error Code Docs: https://www.prisma.io/docs/reference/api-reference/error-reference

`P2013` "Missing the required argument {argument_name} for field {field_name} on {object_name}."

I always get error `P2009`

I do not get a `meta` field with `{ target: [ 'email' ] }` to do my conditional logic on the missing required field.

### How to reproduce

1. Create model with  a required field
2. When creating or updating with create({data:{{}) or update({data:{}}), do not provide the required field. OR provide the required field with a value of `null` or `undefined`
3. Expecting to see error `P2013`, but always get `P2009`,

### Prisma information

```prisma
model  Item{
id Int @id @default(autoincrement())
name String
}
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v16.15.0

### Prisma Version

```
^4.4.0
```

**Comment by arkmech** [comment]
TLDR `P2009` is missing meta and target fields

**Comment by janpio** [comment]
I am not sure I am following. What code are you executing exactly against that schema you shared and what are you getting vs. what are you expecting to get there?

**Comment by arkmech** [comment]
Essentially why doesn't error code `P2009` have a `meta.target` (which is an array of fields) like error code like `P2002`. 

Initially I thought wrong error code was being thrown, but now I think it is correct, just doesn't have the fields I want.

**Comment by janpio** [comment]
Can you please help me understand by sharing a full example:
- Prisma Schema
- Prisma Client Query
- Data in the database (if any that is relevant)
- Full error message you are getting

Right now I am still not fully sure what we are talking about.

Generally P2009 is not an "expected" error from the database as P2002, but something internal that in theory should optimally never happen.

**Comment by janpio** [comment]
Ping @arkmech

**Comment by SevInf** [comment]
@arkmech  We are going to close it since we did not hear back from you. Feel free to open a new issue with relevant issue.
